src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
@MrChrisJohnson not sure if I understand the O(k) algo you're describing

@a1k0n yes as i mentioned the reason you use different trees is to make sure you don't consistently separate the same two close points

but so far annoy has erred too much on the side of uncorrelated trees and too little on using the  distribution of the points. i think you could probably go a little further leveraging the distribution of the points

i'll prob just keep the 2-point solution for now. should be enough